src/zenml/models/hub_plugin_models.py

```diff
@@ -0,0 +1,76 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Models representing ZenML Hub plugins."""
+
+from datetime import datetime
+from typing import TYPE_CHECKING, List, Optional
+from uuid import UUID
+
+from pydantic import BaseModel
+
+from zenml.utils.enum_utils import StrEnum
+
+if TYPE_CHECKING:
+    pass
+
+
+class PluginStatus(StrEnum):
+    """Enum that represents the status of a plugin."""
+
+    PENDING = "pending"  # Plugin is being built
+    FAILED = "failed"  # Plugin build failed
+    AVAILABLE = "available"  # Plugin is available for installation
+    YANKED = "yanked"  # Plugin was yanked and is no longer available
```

Comment:
Can users (and if yes, how? I didn't see a CLI command for that) yank a plugin? Have you thought about the implications of that? Me as the user of a popular plugin can potentially mess with production pipelines of others using that plugin?

Reply:
at the moment, yanking is not possible but we will support this in the future. The idea here would be that yanking simply "hides" this version when listing/browsing plugins, but the wheel itself will still exist